### PR TITLE
feat: add sync bot

### LINKED ISSUE #201: Could there be a more effective way to keep up the Node.js binary updating pace?
Recently I found out that the Node.js binary has arleady released v16.19.1.
image

But the npm packages like node-linux-x64 only contains the v16.19.0.
image

So, is there a better way to keep up the updating pace？Like, automatically publish all the platform packages right after the binaries were released?

### PR DESCRIPTION
**Pull request by elrrrrrrr**
> closes #201, 🤖 Add an automatic synchronization bot via GitHub Workflow.

1. 🔍 Query officially released Node versions in the last 5 days.
2. ♻️ If the corresponding node@x.x.x has not been released, build & release on current branch

🔗 Here is my validation [exp](https://github.com/elrrrrrrr/node-bin-gen/actions/runs/7919031139/job/21619007802) (with dry run)

* 🔐 Project secrets: NPM_TOKEN needs to be configured in this repository.

### CODE REVIEW

**Review by fengmk2 — APPROVED** [review]
+1

**Comment by aredridel** [comment]
I should be able to look at this this week since I'm back from travel.

File: .github/workflows/sync-node-version.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            echo "Checking version $VERSION..."
+            if [ -z "$(npm view node@$VERSION)" ]; then
+              echo "Version $VERSION is not published on npm. Running custom script."
+              mkdir -p "target/$VERSION" && cd "target/$VERSION"
+              node ../../index.js $VERSION
+              for dir in */ ; do
+                if [ -d "$dir" ]; then
+                  echo "Publishing directory: $dir"
+                  cd "$dir"
+                  npm publish
```
Comment (aredridel) [inline .github/workflows/sync-node-version.yml:49]: Can skip the cd, and the if, and `npm publish "$dir/"`

File: .github/workflows/sync-node-version.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                  # npm publish --dry-run
+                  cd ..
+                else
+                  echo "Directory $dir not found."
```
Comment (aredridel) [inline .github/workflows/sync-node-version.yml:53]: This should never happen: you used `*/`

File: .github/workflows/sync-node-version.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          #set -x
+          DATE_5_DAYS_AGO=$(date --date='5 days ago' +%Y-%m-%d)
+          curl -s https://nodejs.org/dist/index.json | jq --arg DATE "$DATE_5_DAYS_AGO" -r '.[] | select(.date >= $DATE) | .version' > versions.txt
+          echo "Node.js versions released in the last 5 days:"
+          cat versions.txt
+          #set +x
+
+      - name: Check each version and sync if not published
+        run: |
+          #set -x
```
Comment (aredridel) [inline .github/workflows/sync-node-version.yml:37]: Probably want a `set -e`

**Comment by aredridel** [comment]
I'll take these changes from here!

**Comment by aredridel** [comment]
Rebased, updated, and merged. This is great work, thank you!

**Comment by aredridel** [comment]
Just coming back to this: this works great, thank you very much! Such a solid contribution.

**Pull request closed without merging**